src/plots/cartesian/layout_attributes.js

```diff
@@ -446,6 +446,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending',
+            'value ascending', 'value descending','array'
+        ],
+        dflt: 'trace',
+        role: 'style',
```

Comment:
`role: 'info'.`

By the way, the attribute `role` fields have no use in plotly.js. They are [stripped](https://github.com/plotly/plotly.js/blob/fb874424d5b89c021ef6e5ebe0326433389d02ee/tasks/util/compress_attributes.js#L31) out on the bundle step. They are only used in [plotly.py](https://plot.ly/python/matplotlib-to-plotly-tutorial/) where they indicate which attribute to strip when the `strip_style` flag is turned on.